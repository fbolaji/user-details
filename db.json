{
  "formConfigData": {
    "userForm": [
      {
        "label": "Name *",
        "id": "name",
        "name": "name",
        "type": "text",
        "placeholder": "Please enter your name",
        "isRequired": true,
        "validator": {
          "isRequired": true,
          "error": "Name is required."
        }
      },
      {
        "label": "Role",
        "id": "role",
        "name": "role",
        "type": "text",
        "placeholder": "Please enter role",
        "isRequired": false,
        "pattern": ""
      },
      {
        "label": "Email *",
        "id": "emailAddress",
        "name": "emailAddress",
        "type": "text",
        "placeholder": "Please enter email address",
        "isRequired": true,
        "pattern": "^[a-zA-Z0-9]+@[a-zA-Z0-9]+\\.[A-Za-z]+$",
        "validator": {
          "isRequired": true,
          "regex": "^[a-zA-Z0-9]+@[a-zA-Z0-9]+\\.[A-Za-z]+$",
          "error": "A valid email address is required."
        }
      },
      {
        "label": "Password *",
        "id": "password",
        "name": "password",
        "type": "password",
        "placeholder": "Please enter password",
        "isRequired": true,
        "pattern": "/(?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).{9,}/",
        "validator": {
          "isRequired": true,
          "regex": "/(?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).{9,}/",
          "error": "Password must be 9 characters long and must contain a lowercase, uppercase and a digit."
        }
      }
    ],
    "userPrivacyForm": [
      {
        "checked": false,
        "label": "Recieve updates about XXXX products by email",
        "id": "productUpdate",
        "name": "productUpdate",
        "type": "checkbox"
      },
      {
        "checked": false,
        "label": "Recieve communication about by email for other products created by XXXX team",
        "id": "otherProductUpdate",
        "name": "otherProductUpdate",
        "type": "checkbox"
      }
    ],
    "userFormValidation": {
      "name": {
        "validator": {
          "isRequired": true,
          "error": "Name is required."
        }
      },
      "email": {
        "validator": {
          "isRequired": true,
          "regex": "\\S+@\\S+\\.\\S+",
          "error": "A valid email address is required."
        }
      },
      "password": {
        "validator": {
          "isRequired": true,
          "regex": "/^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)[A-Za-z\\d@$!%*?&]{10,}$/g",
          "error": "Password must be 9 characters long and must contain a lowercase, uppercase and a digit."
        }
      }
    }
  },
  "usersProfile": []
}