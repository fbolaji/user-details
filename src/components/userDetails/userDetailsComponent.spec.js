import React, {useState as useStateMock} from 'react';
import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import * as rr from 'react-redux';
import { render } from '@testing-library/react';
import UserDetailsComponent from './userDetailsComponent';
import { fetchFormConfigData } from '../../store/actions/user.action';

import * as mockFormConfigData from '../../../db.json';
import rootReducer from '../../store/reducers';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';


const INITIATE_STATE = {
	formConfigData: mockFormConfigData,
	loadingData: false,
	createUserDetails: {},
	error: {},
};
const appReducer = combineReducers({
	rootReducer
});

const composeEnhancers = compose;

const mockStore = createStore(
	appReducer,
	composeWithDevTools(composeEnhancers(applyMiddleware(thunk)))
);

 // jest.mock("react-redux", () => ({
 // 	...jest.requireActual("react-redux"),
 // 	...jest.requireActual("react-router-dom"),
 // 	useSelector: jest.fn(() => mockFormConfigData),
 // 	useStateMock: jest.fn()
 // }));

describe("UserDetailsComponent", () => {
	// afterEach(() => {
	// 	useSelector.mockClear();
	// 	useDispatch.mockClear();
	// });
	it("should render user form", () => {
		// const mockSelector = jest.mock.useSelector;
		const mockDispatch = jest.fn(
			new Promise((resolve, reject) => {
				resolve({
					user: mockFormConfigData
				})
			})
		);

		// const { result } = renderHook(() =>
		// 	UserDetailsComponent(mockDispatch)
		// )
		//expect(result.current.isLoading).toEqual(true)

		//const mockDispatch = jest.fn();
		//jest.spyOn(rr, "useDispatch").mockImplementation(() => mockDispatch);

		const mockFetchConfigData = fetchFormConfigData();
		const { container } = render(<rr.Provider store={mockStore}><UserDetailsComponent /></rr.Provider>);
		//expect(mockDispatch).toHaveBeenCalledTimes(1);
		//expect(container).toMatchSnapshot();
		expect(mockDispatch.call).toBeTruthy();
		expect(container).toMatchSnapshot();
	});
});