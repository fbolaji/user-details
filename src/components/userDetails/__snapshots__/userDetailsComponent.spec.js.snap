// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailsComponent should render user form 1`] = `
<div>
  <div
    class="col-md-8 offset-md-2"
  >
    <div
      class="loader-overlay"
      data-testid="test-loader"
    >
      <div
        class="loader-circle-wrap"
      >
        <div
          class="loader-circle"
        />
      </div>
    </div>
  </div>
</div>
`;
